packages/arb-token-bridge-ui/src/components/common/SidePanel.tsx

```diff
@@ -21,36 +21,92 @@ export const SidePanel = ({
-        <div className="fixed inset-0 bg-dark opacity-80" aria-hidden="true" />
+        <Transition.Child
+          as={Fragment}
+          enter="ease-out duration-[400ms]"
+          enterFrom="opacity-0"
+          enterTo="opacity-80"
+          leave="ease-in duration-300"
+          leaveFrom="opacity-80"
+          leaveTo="opacity-0"
+        >
+          {/* The backdrop, rendered as a fixed sibling to the panel container */}
+          <div className="fixed inset-0 bg-dark" aria-hidden="true" />
+        </Transition.Child>
 
         {/* Full-screen container to center the panel */}
         <div className="fixed inset-0 right-0 top-0 flex h-full w-full items-start justify-end">
-          {/* The heading of dialog  */}
-          <Dialog.Panel
-            className={twMerge(
-              'side-panel flex h-full w-screen max-w-[1000px] flex-col bg-black',
-              panelClassNameOverrides,
-              !scrollable ? '' : 'overflow-y-auto'
-            )}
+          <Transition.Child
+            as={Fragment}
+            enter="ease-out duration-[400ms]"
+            enterFrom="translate-x-full"
+            enterTo="translate-x-0"
+            leave="ease-in duration-300"
+            leaveFrom="translate-x-0"
+            leaveTo="translate-x-full"
+            afterLeave={handleCloseEnd}
           >
-            <Dialog.Title className="sticky top-0 z-50 mx-4 flex flex-row justify-between border-b-[1px] border-gray-6 bg-black py-4 text-white">
-              <span className="text-xl">{heading}</span>
-              <button className="arb-hover" onClick={onClose}>
-                <XMarkIcon className="h-6 w-6 text-white" />
-              </button>
-            </Dialog.Title>
-
-            {/* Contents of the panel */}
-            <div className="side-panel-content z-40 h-full p-4">{children}</div>
-          </Dialog.Panel>
+            {/* The heading of dialog  */}
+            <Dialog.Panel
+              className={twMerge(
+                'side-panel flex h-full w-screen max-w-[1000px] flex-col border-l border-gray-dark bg-black',
+                panelClassNameOverrides,
+                !scrollable ? '' : 'overflow-y-auto'
+              )}
+            >
+              <Dialog.Title className="sticky top-0 z-50 mx-4 flex flex-row justify-between bg-black pt-4 text-white">
+                {heading && <span className="text-2xl">{heading}</span>}
+                <button className="arb-hover" onClick={handleCloseStart}>
+                  <XMarkIcon
+                    className={twMerge(
+                      'h-5 w-5 text-white',
+                      heading ? '' : 'ml-2'
+                    )}
+                  />
+                </button>
+              </Dialog.Title>
+
+              {/* Contents of the panel */}
+              <div
+                className={twMerge(
+                  'side-panel-content z-40 h-full',
+                  heading ? 'p-4' : 'px-4 pb-4'
```

Comment:
might as well put `px-4 pb-4` on the line above and only conditionally add `pt-4`